hikari/impl/shard.py

```diff
@@ -524,6 +524,12 @@ async def get_user_id(self) -> snowflakes.Snowflake:
             raise RuntimeError("user_id was not known, this is probably a bug")
         return self._user_id
 
+    def _get_ws(self) -> _GatewayTransport:
+        if not self._ws:
+            raise errors.ComponentStateConflictError("Shard is shutting down")
+
+        return self._ws
```

Comment:
Would a better approach for this be the WS be raising the `ComponentStateConflictError`?

Then this would leave all the `self._get_ws()` calls to 1 which will only check if the shard has an available ws and use it. Then it is up to the ws to handle it being on or not.

Reply:
That honestly seems like a more complex fix than the one currently present in this PR since it'd probably involve some sort of restructuring

Reply:
Considering we are already using our own `_GatewayTransport`, I dont think it will be that hard, as just all the methods in it will have to check if it alive before executing (as we do in the other components)

Reply:
Keep in mind _GatewayTransport does just extend aiohttp.ClientWebSocketResponse to add some extra logic for handling stuff like serialization, and connection and in itself might already have an error it raises when certain calls are made and it's considered closed

Reply:
Considering it is private implementation and we only make use of the methods in _GatewayTransport, we could add the validation there

Reply:
And then you've also gotta refactor how the Shard object handles the ws and add stuff like is alive methods to the gateway transport etc etc and check to see whether its reusable to see if these refactors should be replacing it on restart or not and if it is reusable we'll also have to add a start method to it and it's all just a lot of extra effort for no added benefit